pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -674,7 +670,14 @@ stdenv.mkDerivation {
   preFixup = lib.optionalString withEfi ''
     mv $out/lib/systemd/boot/efi $out/dont-strip-me
   '';
-  postFixup = lib.optionalString withEfi ''
+
+  postFixup = ''
+    # Wrap in the correct path for LUKS2 tokens. Must be after the fixup phase
+    # or the rpath cleanup removes the directories again.
```

Comment:
This comment doesn't look right. Down there, we set `LD_LIBRARY_PATH`, not set rpath. Also, I'd propose moving the comment above `postFixup`, so iterating on the comment itself doesn't trigger rebuilds.